app/src/main/java/org/mozilla/fenix/settings/address/view/AddressEditorState.kt

```diff
@@ -0,0 +1,51 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.settings.address.view
+
+import mozilla.components.browser.state.state.BrowserState
+import mozilla.components.concept.storage.Address
+import org.mozilla.fenix.settings.address.AddressEditorFragment
+import org.mozilla.fenix.settings.address.AddressEditorFragment.Companion.availableCountries
+import org.mozilla.fenix.settings.address.AddressEditorFragment.Companion.defaultCountry
+
+/**
+ * The state for the [AddressEditorFragment].
+ *
+ * @property country The currently selected [AddressCountry].
+ */
+data class AddressEditorState(
```

Comment:
I believe we can avoid having `AddressEditorState` altogether, and just add all the country values and the default country handling inside of the `AddressEditorView`. We had an `CreditCardEditorState` because I needed a new data type that translated `CreditCard` into something that made sense for the UI - credit card number needed to be decrypted and values needed to be casted, etc. 

I think we can just handle this in the view as we are binding the dropdown values. This also feels cleaner to me instead of having the country values as companion values in the fragment, and having a state data class that only handles the country value. I think we got lucky with `Address` because it didn't need any transformation and we can simply bind and set the text so we just pass it as a parameter into the view.

Reply:
Yeah, that makes sense to me. Thanks for the suggestion